client/lib/media/actions.js

```diff
@@ -37,7 +37,11 @@ MediaActions.setQuery = function( siteId, query ) {
 	} );
 };
 
-MediaActions.fetch = function( siteId, itemId ) {
+// TODO: Once this entire module is converted to Redux,
+// `receiveMediaItems` shouldn't be passed as an argument anymore,
+// but directly imported and dispatch()ed from inside `fetch()`,
+// which needs to be turned into a Redux thunk.
+MediaActions.fetch = function( siteId, itemId, receiveMediaItems = null ) {
```

Comment:
Related to recent discussions, I think if we're to introduce compatibility, I'd rather it look like #2167 than as a callback to an existing Flux implementation (which, as in Redux, is poor data flow). It would simplify this pull request a ton and help avoid cluttering up existing components and Flux actions with this argument passing.